File: app/src/main/java/com/cogu/spylook/model/relationships/CausanteSuceso.java
package com.cogu.spylook.model.relationships;

import androidx.room.Embedded;
import androidx.room.Relation;

import com.cogu.spylook.model.entity.Contacto;
import com.cogu.spylook.model.entity.Suceso;

import java.util.List;

public class CausanteSuceso {
    @Embedded
    public Contacto contacto;

    @Relation(parentColumn = "id", entityColumn = "idCausante")
    public List<Suceso> sucesos;
}

File: app/src/main/java/com/cogu/spylook/model/relations/CausanteSuceso.kt
package com.cogu.spylook.model.relations

import androidx.room.Embedded
import androidx.room.Relation
import com.cogu.spylook.model.entity.Contacto
import com.cogu.spylook.model.entity.Suceso

class CausanteSuceso {
    @Embedded
    var contacto: Contacto? = null

    @Relation(parentColumn = "id", entityColumn = "idCausante")
    var sucesos: MutableList<Suceso?>? = null
}
